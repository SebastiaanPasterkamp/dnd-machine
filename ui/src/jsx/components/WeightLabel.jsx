import React from 'react';
import PropTypes from 'prop-types';

const WeightLabel = function(props) {
    const { weight } = props;
    const { lb, oz } = weight === null ? props : {
        lb: Math.floor(weight),
        oz: Math.floor((weight % 1) * 16)
    };

    if (lb === null && oz === null) {
        return null;
    }
    return (
        <div className="weight inline">
            {lb ? (<span>{ lb }lb</span>) : null}
            {oz ? (<span>{ oz }oz</span>) : null}
        </div>
    );
}

WeightLabel.propTypes = {
    weight: PropTypes.number,
    lb: PropTypes.number,
    oz: PropTypes.number,
};

WeightLabel.defaultProps = {
    weight: null,
    lb: null,
    oz: null,
};

export default WeightLabel;
