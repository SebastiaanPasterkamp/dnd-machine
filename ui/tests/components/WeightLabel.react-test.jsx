import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import WeightLabel from 'components/WeightLabel';

describe('WeightLabel', () => {
    it('should render nothing without props', () => {
        const tree = renderer.create(
            <WeightLabel />
        );

        expect(tree).toMatchSnapshot();
    });

    describe('with explicity lb and/or oz', () => {
        it('should render only lb', () => {
            const tree = renderer.create(
                <WeightLabel lb={1} />
            );

            expect(tree).toMatchSnapshot();
        });

        it('should render only oz', () => {
            const tree = renderer.create(
                <WeightLabel oz={2} />
            );

            expect(tree).toMatchSnapshot();
        });

        it('should render both', () => {
            const tree = renderer.create(
                <WeightLabel lb={3} oz={4} />
            );

            expect(tree).toMatchSnapshot();
        });
    });

    describe('with computed weight', () => {
        it('should render only lb', () => {
            const tree = renderer.create(
                <WeightLabel weight={1} />
            );

            expect(tree).toMatchSnapshot();
        });

        it('should render only oz', () => {
            const tree = renderer.create(
                <WeightLabel weight={0.125} />
            );

            expect(tree).toMatchSnapshot();
        });

        it('should render both', () => {
            const tree = renderer.create(
                <WeightLabel weight={3.25} />
            );

            expect(tree).toMatchSnapshot();
        });
    });
});
