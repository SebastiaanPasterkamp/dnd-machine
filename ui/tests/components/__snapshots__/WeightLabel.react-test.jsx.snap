// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeightLabel should render nothing without props 1`] = `null`;

exports[`WeightLabel with computed weight should render both 1`] = `
<div
  className="weight inline"
>
  <span>
    3
    lb
  </span>
  <span>
    4
    oz
  </span>
</div>
`;

exports[`WeightLabel with computed weight should render only lb 1`] = `
<div
  className="weight inline"
>
  <span>
    1
    lb
  </span>
</div>
`;

exports[`WeightLabel with computed weight should render only oz 1`] = `
<div
  className="weight inline"
>
  <span>
    2
    oz
  </span>
</div>
`;

exports[`WeightLabel with explicity lb and/or oz should render both 1`] = `
<div
  className="weight inline"
>
  <span>
    3
    lb
  </span>
  <span>
    4
    oz
  </span>
</div>
`;

exports[`WeightLabel with explicity lb and/or oz should render only lb 1`] = `
<div
  className="weight inline"
>
  <span>
    1
    lb
  </span>
</div>
`;

exports[`WeightLabel with explicity lb and/or oz should render only oz 1`] = `
<div
  className="weight inline"
>
  <span>
    2
    oz
  </span>
</div>
`;
